src/Compilers/VisualBasic/Test/Emit/Emit/EmitMetadata.vb

```diff
@@ -973,5 +975,104 @@ End Class
+            CompileAndVerify(
+                <compilation>
+                    <file>
+Module Program
+    Sub Main()
+        System.Console.WriteLine("a")
+    End Sub
+End Module
+                    </file>
+                </compilation>,
+                options:=TestOptions.ReleaseExe.WithMetadataImportOptions(MetadataImportOptions.All),
+                emitOptions:=EmitOptions.Default.WithEmitMetadataOnly(True),
+                symbolValidator:=Sub(m)
+                                     Assert.NotEqual(0, m.GetMetadata().Module.PEReaderOpt.PEHeaders.CorHeader.EntryPointTokenOrRelativeVirtualAddress)
+                                     Assert.NotNull(m.GlobalNamespace.GetMember(Of MethodSymbol)("Program.Main"))
+                                 End Sub
+            ).VerifyDiagnostics()
+        End Sub
+
+        <Fact, WorkItem("https://github.com/dotnet/roslyn/issues/76707")>
+        Public Sub EmitMetadataOnly_Exe_NoMain()
+            Dim emitResult = CreateCompilation(
+                <compilation>
+                    <file>
+Module Program
+End Module
+                    </file>
+                </compilation>,
+                options:=TestOptions.ReleaseExe,
+                assemblyName:="MyLib"
+            ).Emit(New MemoryStream(), options:=EmitOptions.Default.WithEmitMetadataOnly(True))
+            Assert.False(emitResult.Success)
+            emitResult.Diagnostics.AssertTheseDiagnostics(<errors>
+BC30420: 'Sub Main' was not found in 'MyLib'.
+                                                          </errors>)
+        End Sub
+
+        <Fact, WorkItem("https://github.com/dotnet/roslyn/issues/76707")>
+        Public Sub EmitMetadataOnly_Exe_PrivateMain_ExcludePrivateMembers()
+            CompileAndVerify(
+                <compilation>
+                    <file>
+Module Program
+    Friend Sub Main()
+    End Sub
+End Module
+                    </file>
+                </compilation>,
+                options:=TestOptions.ReleaseExe.WithMetadataImportOptions(MetadataImportOptions.All),
+                emitOptions:=EmitOptions.Default.WithEmitMetadataOnly(True).WithIncludePrivateMembers(False),
+                symbolValidator:=Sub(m)
+                                     Assert.NotEqual(0, m.GetMetadata().Module.PEReaderOpt.PEHeaders.CorHeader.EntryPointTokenOrRelativeVirtualAddress)
+                                     Assert.NotNull(m.GlobalNamespace.GetMember(Of MethodSymbol)("Program.Main"))
+                                 End Sub
+            ).VerifyDiagnostics()
+        End Sub
+
+        <Fact, WorkItem("https://github.com/dotnet/roslyn/issues/76707")>
+        Public Sub ExcludePrivateMembers_PrivateMain()
+            Using peStream As New MemoryStream()
+                Using metadataStream As New MemoryStream()
+                    Dim comp = CreateCompilation(
+                        <compilation>
+                            <file>
+Module Program
+    Friend Sub Main()
```

Comment:
>Friend [](http://example.com/codeflow?start=4&length=6)

Private?

Reply:
That doesn't work in VB. I guess I can rename the test though.